### PR TITLE
Failure message is nested text now: allows to see newlines in Jenkins…

### PR DESCRIPTION
**Pull request by KidsKilla**
… reports

### CODE REVIEW

File: utils/buildJsonResults.js [hunk 1]
```diff
@@ -64,11 +65,7 @@ module.exports = function (report, appDirectory, options) {
       if (tc.status === 'failed') {
         tc.failureMessages.forEach((failure) => {
           testCase.testcase.push({
-            'failure': [{
-              _attr: {
-                message: failure || ''
-              }
-            }]
+            'failure': stripAnsi(failure || '')
```
Comment (palmerj3) [inline utils/buildJsonResults.js]: The "_attr" attribute needs to be there otherwise the format of the junit.xml won't be to spec

**Comment by palmerj3** [comment]
Please remove the yarn.lock file. For libs it's best not to include it. Yarn ignores dependencies yarn.lock files anyways and it's best to not give the impression that deps are locked down for a lib. You lock down dependencies for applications, not libs.

**Comment by palmerj3** [comment]
I'm a little worried about the use of strip-ansi.. the bit I read about it it is actually removing characters. So I'm not sure how this enables you to see newlines? If you could explain that a bit further I would appreciate. Thanks!

**Comment by yangshun** [comment]
I'm facing a problem where I'm unable to open the exported XML file in Chrome browser. My file contents can be found [here](https://uploadfiles.io/cfdtf).

I believe that the stripping ansi feature will solve the issue I'm facing too.

**Comment by palmerj3** [comment]
I see - I think the issue here is actually that the stack trace should be within the <failure></failure> tags not within a property of the failure tag. I'll work on a separate PR to adjust that.

Going to close this.

I do really appreciate the PR but I want to avoid stripping out data if possible since I think that could break other implementations.

**Comment by KidsKilla** [comment]
Hi! Thanks for response.

> I see - I think the issue here is actually that the stack trace should be within the tags not within a property of the failure tag. I'll work on a separate PR to adjust that.

Mainly, yes.

> I do really appreciate the PR but I want to avoid stripping out data if possible since I think that could break other implementations.

Than couldn't, actually. The problem here is because xml is like tty. But it introduces problems since xml contents usually displays as-is, so [Escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code) simply introduces bugs like above. I've added it myself because Jenkins simply shows them, ruining readability.

No worries, stripAnsi is written by author of well-known npm module: [chalk](https://www.npmjs.com/package/chalk), it doesn't do any unknown untested crap.

**Comment by KidsKilla** [comment]
As for yarn.lock: I've updated PR, just in case.

**Comment by palmerj3** [comment]
I created another PR which accomplishes what you were looking to do. Even included strip-ansi now that I understand a bit more what it does. I was able to use the modified code in an example project, throw an error in a test case, and get the kinds of results you showed me.

Please see https://github.com/palmerj3/jest-junit/pull/21

@KidsKilla @yangshun

**Pull request closed without merging**